packages/sanity/src/core/bundles/components/dialog/BundleDetailsDialog.tsx

```diff
@@ -0,0 +1,120 @@
+import {ArrowRightIcon} from '@sanity/icons'
+import {Box, Button, Dialog, Flex} from '@sanity/ui'
+import {type FormEvent, useCallback, useState} from 'react'
+
+import {type BundleDocument} from '../../../store/bundles/types'
+import {useBundleOperations} from '../../../store/bundles/useBundleOperations'
+import {usePerspective} from '../../hooks/usePerspective'
+import {BundleForm} from './BundleForm'
+
+interface BundleDetailsDialogProps {
+  onCancel: () => void
+  onSubmit: () => void
+  bundle?: BundleDocument
+}
+
+export function BundleDetailsDialog(props: BundleDetailsDialogProps): JSX.Element {
+  const {onCancel, onSubmit, bundle} = props
+  const {createBundle, updateBundle} = useBundleOperations()
+  const [hasErrors, setHasErrors] = useState(false)
+  const formAction = bundle ? 'edit' : 'create'
+
+  const [value, setValue] = useState<Partial<BundleDocument>>(() => {
+    if (bundle) {
+      return {
+        slug: bundle.slug,
+        title: bundle.title,
+        description: bundle.description,
+        hue: bundle.hue,
+        icon: bundle.icon,
+      }
+    }
+
+    return {
+      slug: '',
+      title: '',
+      hue: 'gray',
+      icon: 'cube',
+      //publishAt: undefined,
+    }
+  })
+  const [isSubmitting, setIsSubmitting] = useState(false)
+
+  // TODO MAKE SURE THIS IS HOW WE WANT TO DO THIS
+  const {setPerspective} = usePerspective()
+
+  const bundleOperation = useCallback(
+    (formValue: Partial<BundleDocument>) => {
+      if (bundle) {
+        const updatedBundle: BundleDocument = {
+          ...bundle,
```

Comment:
I think.. let's confirm it, that you don't need to send the bundle document, given the way the set action works it will only set the keys provided, so in theory, we could provide only the form value

Reply:
I tested this and, yep, only need to send across the edited fields. So I've updated this now